app/controllers/NirmsNumberController.scala

```diff
@@ -40,30 +40,35 @@ class NirmsNumberController @Inject() (
     with I18nSupport {
   private val form = formProvider()
 
-  def onPageLoad: Action[AnyContent] = (authorise andThen sessionRequest) { implicit request =>
+  def onPageLoad(mode: Mode): Action[AnyContent] = (authorise andThen sessionRequest) { implicit request =>
     val optionalNirmsNumber = request.userAnswers.maintainProfileAnswers.nirmsNumber
 
     optionalNirmsNumber match {
-      case Some(nirmsNumber) => Ok(view(form.fill(nirmsNumber.value)))
-      case None              => Ok(view(form))
+      case Some(nirmsNumber) => Ok(view(form.fill(nirmsNumber.value), mode))
+      case None              => Ok(view(form, mode))
     }
   }
 
-  def onSubmit: Action[AnyContent] = (authorise andThen sessionRequest).async { implicit request =>
+  def onSubmit(mode: Mode): Action[AnyContent] = (authorise andThen sessionRequest).async { implicit request =>
     form
       .bindFromRequest()
       .fold(
-        formWithErrors => Future.successful(BadRequest(view(formWithErrors))),
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
         nirmsNumber => {
           val updatedMaintainProfileAnswers =
-            request.userAnswers.maintainProfileAnswers.copy(nirmsNumber = Some(NirmsNumber(nirmsNumber)))
+            request.userAnswers.maintainProfileAnswers
+              .copy(nirmsNumber = Some(NirmsNumber(nirmsNumber)))
           val updatedUserAnswers            = request.userAnswers.copy(maintainProfileAnswers = updatedMaintainProfileAnswers)
 
           sessionService
             .updateUserAnswers(updatedUserAnswers)
             .fold(
               sessionError => Redirect(routes.JourneyRecoveryController.onPageLoad().url),
-              success => Redirect(routes.NiphlQuestionController.onPageLoad.url)
+              success =>
+                mode match {
```

Comment:
Might be nice to put this in its own private function in the controller
`private def navigate(mode: Mode) = mode match....`

Reply:
done and added this to thje other controllers too - i agree it looks nicer